calicoctl/commands/node/run.go

```diff
@@ -526,9 +535,28 @@ func validateIpAutodetectionMethod(method string, version int) {
 		// Auto-detection method is "interface", validate that the interface
 		// regex is a valid golang regex.
 		ifStr := strings.TrimPrefix(method, AUTODETECTION_METHOD_INTERFACE)
-		if _, err := regexp.Compile(ifStr); err != nil {
-			fmt.Printf("Error executing command: invalid interface regex specified for IP autodetection: %s\n", ifStr)
-			os.Exit(1)
+
+		// Regexes are provided in a string separated by ","
+		ifRegexes := strings.Split(ifStr, ",")
+		for _, ifRegex := range ifRegexes {
+			if _, err := regexp.Compile(ifStr); err != nil {
+				fmt.Printf("Error executing command: invalid interface regex specified for IP autodetection: %s\n", ifRegex)
+				os.Exit(1)
+			}
+		}
+		return
+	} else if strings.HasPrefix(method, AUTODETECTION_METHOD_SKIP_INTERFACE) {
+		// Auto-detection method is "skip-interface", validate that the
+		// interface regexes used are valid golang regexes.
+		ifStr := strings.TrimPrefix(method, AUTODETECTION_METHOD_SKIP_INTERFACE)
+
+		// Regexes are provided in a string separated by ","
+		ifRegexes := strings.Split(ifStr, ",")
+		for _, ifRegex := range ifRegexes {
+			if _, err := regexp.Compile(ifRegex); err != nil {
+				fmt.Printf("Error executing command: invalid interface skipping regex specified for IP autodetection: %s\n", ifRegex)
```

Comment:
nit: can we reword to remove the word "skipping" from this?

I think `invalid interface regex` gets the point across well enough - as is it sounds like it's going to skip the invalid regex, when actually it's just exiting.